### ISSUE TITLE
build issues.

### ISSUE THREAD

**Issue by jackli577** [virtio-win/kvm-guest-drivers-windows#159]
when i build with cmd. and aslo install vc 2005 and wdk.  
2>  Setting OS mask for: Wxp x86 to XP_X86,Server2003_X86
2>  ............................
2>  Signability test failed.
2>
2>  Errors:
2>  22.9.6: DriverVer missing or in incorrect format in \balloon.inf

use vc, it ok.

can give some help.. 

and aslo  i found some tools  blnsrv or netkvmco,  use cmd to build will not find lib. result to buildall stop.  but driver in build out.

**Comment by vrozenfe** [comment]
can it be that you don't have wdk 7.1 (https://download.microsoft.com/download/4/A/2/4A25C7D5-EFBE-4182-B6A9-AE6850409A78/GRMWDK_EN_7600_1.ISO) installed on your build system? or its location is different from the default (c:\winddk\7600.16385.1\)

**Comment by jackli577** [comment]
@vrozenfe   thanks in advance,  i install wdk in right version and path.
i comment <Exec Command="inf2cat /driver:$(PackOne_DestinationPath) /os:$(PackOne_OsMask)" />
ok. but inf no driverver

**Comment by vrozenfe** [comment]
So you are still hitting the same problem with .inf time stamping as mentioned before
2> 22.9.6: DriverVer missing or in incorrect format in \balloon.inf
or it is something else?

Could you please post the entire build log output?
Thanks,
Vadim.

**Comment by YanVugenfirer** [comment]
Please check line endings. Stamp INF failed on Unix style EOL.

Please check wiki: http://www.linux-kvm.org/page/WindowsGuestDrivers/Git_and_crlf

**Comment by jackli577** [comment]
@vrozenfe  @YanVugenfirer  i find the problem is:  when i use vs to build,  the stampinf is use ddk direction,  use cmd, it use wdk diretion.

vs: C:\WinDDK\7600.16385.1\bin\x86\stampinf.exe
cmd: C:\Program Files (x86)\Windows Kits\10\bin\x64\stampinf.exe

cmd also when build xp x86,  use wdk  x64 stampinf.exe

**Comment by YanVugenfirer** [comment]
Please provide build logs as Vadim mentioned. Thanks!

**Comment by jackli577** [comment]
ok,  sorry.

2>  balloon.c
2>  VZ_RELEASE_N=VZ_RELEASE_N
2>  VZ_RELEASE_A=VZ_RELEASE_A
2>  VZ_RELEASE_B=VZ_RELEASE_B
2>  VZ_RELEASE_C=VZ_RELEASE_C
2>  _NT_TARGET_MAJ=51
2>  _RHEL_RELEASE_VERSION_=6
2>  _BUILD_MAJOR_VERSION_=101
2>  _BUILD_MINOR_VERSION_=58000
2>  RHEL_COPYRIGHT_STARTING_YEAR=2009
2>  VZ_COPYRIGHT_STARTING_YEAR=VZ_COPYRIGHT_STARTING_YEAR
2>  NTDDI_VERSION=0x05010200
2>  Device.c
2>  Driver.c
2>  memstat.c
2>  queue.c
2>  utils.c
2>  正在生成代码...
2>  balloon.vcxproj -> C:\work\tem\kvm-guest-drivers-windows\Balloon\sys\objfre_wxp_x86\i386\balloon.sys
2>  balloon.vcxproj -> objfre_wxp_x86\i386\balloon.pdb (Full PDB)
2>  AdjustInf_LegacyDDK: for balloon.inx to objfre_wxp_x86\i386\balloon.inf
2>  Adding INX_COPYRIGHT_1=>Copyright (c) 2009-2017 Red Hat Inc.
2>  Adding INX_COPYRIGHT_2=>
2>  Adding INX_COMPANY=>Red Hat, Inc.
2>  Adding INX_PREFIX_VENDOR=>Red Hat
2>  Adding INX_PREFIX_VIRTIO=>
2>  Adding INX_PREFIX_QEMU=>
2>  Processing file C:\work\tem\kvm-guest-drivers-windows\Balloon\sys\objfre_wxp_x86\i386\balloon.inf
2>  Saving original context to: C:\work\tem\kvm-guest-drivers-windows\Balloon\sys\objfre_wxp_x86\i386\balloon.inf.orig
2>  Saving replaced context to: C:\work\tem\kvm-guest-drivers-windows\Balloon\sys\objfre_wxp_x86\i386\balloon.inf.new
2>  StampInf_LegacyDDK: for objfre_wxp_x86\i386\balloon.inf with Feature_UsingWDF=true and STAMPINF_VERSION=51.6.101.58000
2>  Stamping objfre_wxp_x86\i386\balloon.inf [Version] section with DriverVer=
2>  Checking PackOne: TargetOS=Wxp and TargetArch=x86
2>  PackOneTarget Wxp x86 balloon
2>  Copying PackOne_Files=..\app\objfre_wxp_x86\i386\blnsvr.exe to ..\Install\Wxp\x86
2>  Copying PackOne_Files=..\app\objfre_wxp_x86\i386\blnsvr.pdb to ..\Install\Wxp\x86
2>  Copying PackOne_Files=objfre_Wxp_x86\i386\balloon.sys to ..\Install\Wxp\x86
2>  Copying PackOne_Files=objfre_Wxp_x86\i386\balloon.pdb to ..\Install\Wxp\x86
2>  Copying PackOne_Files=objfre_Wxp_x86\i386\balloon.inf to ..\Install\Wxp\x86
2>  Copying PackOne_Files=C:\WINDDK\7600.16385.1\redist\wdf\x86\WdfCoInstaller01009.dll to ..\Install\Wxp\x86
2>  Setting OS mask for: Wxp x86 to XP_X86,Server2003_X86
2>  ............................
2>  Signability test failed.
2>
2>  Errors:
2>  22.9.6: DriverVer missing or in incorrect format in \balloon.inf
2>
2>  Warnings:
2>  None
2>C:\work\tem\kvm-guest-drivers-windows\Tools\Driver.PackOne.targets(180,5): error MSB3073: 命令“inf2cat /driver:..\Install\Wxp\x86 /os:XP_X86,Server2003_X86”已退出，代码为 -2。
========== 生成: 成功 1 个，失败 1 个，最新 2 个，跳过 0 个 ==========

**Comment by ladipro** [comment]
The stampinf tool appears to have a bug in recent Win10 WDK versions. Sometimes it generates garbage if -d is not specified on the command line.

Can you please try this patch:

```
--- a/Tools/Driver.Common.targets
+++ b/Tools/Driver.Common.targets
@@ -103,8 +103,8 @@ Invoking the task:
   <!-- Stampinf (for Legacy DDK) -->
   <Target Condition="'$(Feature_LegacyStampInf)'=='true' AND '$(UseLegacyDDK)'=='true'" Name="StampInf_LegacyDDK" AfterTargets="AdjustInf_LegacyDDK" BeforeTargets="PackOneTarget">
     <Message Text="StampInf_LegacyDDK: for $(OutDir)$(TargetName).inf with Feature_UsingWDF=$(Feature_UsingWDF) and STAMPINF_VERSION=$(STAMPINF_VERSION)" Importance="high"/>
-    <Exec Condition="'$(Feature_UsingWDF)'=='false'" Command="stampinf -f $(OutDir)$(TargetName).inf -a $(InfArch) -v $(STAMPINF_VERSION)"/>
-    <Exec Condition="'$(Feature_UsingWDF)'=='true'" Command="stampinf -f $(OutDir)$(TargetName).inf -a $(InfArch) -k 1.9 -v $(STAMPINF_VERSION)"/>
+    <Exec Condition="'$(Feature_UsingWDF)'=='false'" Command="stampinf -f $(OutDir)$(TargetName).inf -a $(InfArch) -v $(STAMPINF_VERSION) -d *"/>
+    <Exec Condition="'$(Feature_UsingWDF)'=='true'" Command="stampinf -f $(OutDir)$(TargetName).inf -a $(InfArch) -k 1.9 -v $(STAMPINF_VERSION) -d *"/>
   </Target>

   <!-- Imports -->


```

**Comment by jackli577** [comment]
@ladipro   thanks .  it is nice.  yesterday i saw u changed  netkvm in log.  try it ok.

SHA-1: cadc8743496c9a86f7dcddc68d18b7e47c5b5c97

* netkvm: Fix build break with WDK 10.0.15063.0

stampinf.exe in the latest Win10 WDK appears to have a bug and writes
out an empty DriverVer= directive when passed a command line like below:

  stampinf -f <inf> -a <arch> -v <version>

Adding -d seems to work around it:

  stampinf -f <inf> -a <arch> -v <version> -d *

Signed-off-by: Ladi Prosek <lprosek@redhat.com>
